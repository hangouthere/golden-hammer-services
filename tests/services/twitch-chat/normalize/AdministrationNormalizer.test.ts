import normalizerClass from '-/twitch-chat/normalize/AdministrationNormalizer';

const MOCK_DATA = {
  duration: 99,
  userState: {
    'tmi-sent-ts': '1649705365314',
    'target-user-id': 'targetUserId'
  }
};

describe('Twitch Chat: Normalizer - Administration', () => {
  let normalizer;

  beforeEach(() => {
    normalizer = new normalizerClass();
  });

  it('should normalize a Timeout', () => {
    const normalized = normalizer.normalizeTimeout({
      incomingEventArguments: ['userTimedout', null, MOCK_DATA.duration, MOCK_DATA.userState]
    });

    expect(normalized).toMatchSnapshot();
  });

  it('should normalize a Ban', () => {
    const normalized = normalizer.normalizeBan({
      incomingEventArguments: ['userBanned', null, MOCK_DATA.userState]
    });

    expect(normalized).toMatchSnapshot();
  });

  it('should normalize a Message Removal', () => {
    const normalized = normalizer.normalizeMessageRemoval({
      incomingEventArguments: [
        'userMessageRemoved',
        'removed message',
        { ...MOCK_DATA.userState, 'target-msg-id': 'targetMessageIdForRemoval' }
      ]
    });

    expect(normalized).toMatchSnapshot();
  });
});
