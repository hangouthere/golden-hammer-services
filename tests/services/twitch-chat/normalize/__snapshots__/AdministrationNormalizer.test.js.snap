// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twitch Chat: Normalizer - Administration should normalize a Ban 1`] = `
Object {
  "normalizedData": Object {
    "targetId": "targetUserId",
    "userName": "userBanned",
  },
  "timestamp": 1649705365314,
}
`;

exports[`Twitch Chat: Normalizer - Administration should normalize a Message Removal 1`] = `
Object {
  "normalizedData": Object {
    "removedMessage": "removed message",
    "targetId": "targetMessageIdForRemoval",
    "userName": "userMessageRemoved",
  },
  "timestamp": 1649705365314,
}
`;

exports[`Twitch Chat: Normalizer - Administration should normalize a Timeout 1`] = `
Object {
  "normalizedData": Object {
    "duration": 99,
    "targetId": "targetUserId",
    "userName": "userTimedout",
  },
  "timestamp": 1649705365314,
}
`;
